src/display/edition/freetext.js

```diff
@@ -0,0 +1,225 @@
+/* Copyright 2022 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AnnotationEditorType, Util } from "../../shared/util.js";
+import { AnnotationEditor } from "./editor.js";
+import { bindEvents } from "./tools.js";
+
+/**
+ * Basic text editor in order to create a FreeTex annotation.
+ */
+class FreeTextEditor extends AnnotationEditor {
+  #color;
+
+  #content = "";
+
+  #contentHTML = "";
+
+  #fontSize;
+
+  static _freeTextDefaultContent = "";
+
+  constructor(params) {
+    super({ ...params, name: "freeTextEditor" });
+    this.#color = params.color || "CanvasText";
+    this.#fontSize = params.fontSize || 10;
+  }
+
+  static setL10n(l10n) {
+    l10n?.get("freetext_default_content").then(msg => {
```

Comment:
Please add this new string to the list in https://github.com/mozilla/pdf.js/blob/master/web/l10n_utils.js, to ensure that a fallback value is always available.

Furthermore, is there ever a risk that the localized string becomes available *after* the attribute has been set below?

Reply:
Good question: I asked it to myself.
So `l10n` is queried at the first layer creation, in order to see it (or not) the user need to click on the text editor button then move their mouse somewhere on the page, click and then the div is created with an attribute having (or not) the localized value. My feeling is it's very unlikely possible to have a wrong string. I thought to try to implement something in js to emulate the `empty::before` behaviour, but I wasn't super excited by the idea... compared the easy solution we've right now.
Anyway, as usual if you have a good idea, I take it !

Reply:
Would it work if the attribute is always updated asynchronously instead, or does that not work?

Because in that case we could just store the Promise here, rather than the raw string, with something like
```javascript
this._defaultContentPromise = l10n.get("freetext_default_content");
```
and then set the attribute with
```javascript
FreeTextEditor._defaultContentPromise.then(msg => {
  this.editorDiv.setAttribute("default-content", msg);
});
```

Reply:
I tested here:
https://jsfiddle.net/ub7tmxdo/13/
and it works.